source/leuven/wice_advanced_guide.rst

```diff
@@ -0,0 +1,121 @@
+.. _wice_t2_leuven_advanced:
+
+===================
+wICE advanced guide
+===================
+
+.. _wice_compilation:
+
+Compiling software
+------------------
+
+The wICE compute nodes feature Intel processors nicknamed IceLake, which are
+the successors of the SkyLake and CascadeLake architectures which you can find
+on Genius. Although architectural differences are rather small, it is highly
+recommended to compile a version of your code specifically for wICE. Since the
+operating system (Rocky Linux 8.6) differs significantly from Genius, a
+separate compilation is strictly required in most cases anyway. A good approach
+to compile on a wICE node is to launch an interactive job.
+
+Many dependencies you might need are centrally installed. The modules that are
+optimized for wICE are available in ``/apps/leuven/icelake/2021a/modules/all``.
```

Comment:
Again, hard-coded toolchain version.
We better not give the impression that there is and will be only `2021a` toolchain available on wICE, but rather newer toolchains will keep coming as the time lapses. So, you may add/modify this line like this:
"The modules that are optimized for wICE are available under the default path ``/apps/leuven/icelake/2021a/modules/all``. More recent modules will be compiled using more modern toolchains `<toolchain-year>` and will be regularly made available under the path `/apps/leuven/icelake/<toolchain-year>/modules/all`." In order to swap between modules from different `<toolchain-year>`, you may use the `module use` command, e.g. 
```
module use /apps/leuven/icelake/<toolchain-year>/modules/all
```
The valid choices for `<toolchain-year>` look like `2021a`, `2022b`, etc.

Reply:
Some of the things you mention here were already in the text, but just a bit further down. I made some adaptations to it based on the comments. While I agree that in general it is a good idea to avoid specifying versions that can change, I don't think this is very relevant for the toolchain version. `2021a` will normally remain the default one on wICE for its lifetime. If we add a toolchain version that gives a significant advantage, we can adapt the documentation.